### PR TITLE
Fix unit test script to return test command exit code

### PR DESCRIPTION
**Pull request by amitkma**
### 🎯 Goal

To set GitHub action's status according to Gradle tests command output.

### 🛠 Implementation details

Returned from the script with exit code of tests command. 

### 🧪 Testing

Make sure GitHub action fails for failing unit tests. 

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created))
- [ ] Reviewers added

### CODE REVIEW

File: stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/integration/BaseConnectedIntegrationTest.kt [hunk 1]
```diff
@@ -71,10 +71,10 @@ internal open class BaseConnectedIntegrationTest : BaseDomainTest() {
             backgroundSyncEnabled,
             context,
             offlinePlugin = plugin,
+            retryPolicy = NoRetryPolicy()
```
Comment (JcMinarro) [inline stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/integration/BaseConnectedIntegrationTest.kt:74]: Why this change is needed?

File: scripts/ci-unit-tests.sh [hunk 2]
```diff
@@ -4,4 +4,10 @@ command_file="unit-tests-command.sh"
 chmod +x $command_file && \
 ./$command_file
 
+# Exit code of gradle test command output
+exitCode=$?
+
 rm $command_file
+
+# Exit with test command output to set the GH action's check run status
+exit $exitCode
```
Comment (JcMinarro) [inline scripts/ci-unit-tests.sh:13]: We don't really need to store the previous command exit code. 
As far as we are concatenated command on a single line and using `&&`, whenever a command doesn't return `0` (It was run properly without errors), the following commands are not run
```suggestion
./$command_file && \
rm $command_file
```

**Review by JcMinarro — CHANGES_REQUESTED** [review]

File: scripts/ci-unit-tests.sh (hunk 2 above)
Comment (amitkma) [reply scripts/ci-unit-tests.sh:13]: Ah good suggestions. Then the last command to delete the temporary `unit-tests-command` won't be deleted. Will it be a problem?

Comment (JcMinarro) [reply scripts/ci-unit-tests.sh:13]: It shouldn't. It will be there on the docker machine that is created, but because on every CI-Job the docker machine is a new one, it is not going to affect

File: stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/integration/BaseConnectedIntegrationTest.kt (hunk 1 above)
Comment (amitkma) [reply stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/integration/BaseConnectedIntegrationTest.kt:74]: Not needed anymore as it will be done by Oleg's PR.

**Review by JcMinarro — APPROVED** [review]

**Review by ogkuzmin — APPROVED** [review]
